.travis.yml

```diff
@@ -19,6 +19,7 @@ install:
   - make travis-install
 script:
   - make check_warnings
```

Comment:
A counterpart for this could be added for the `eunit` tests to compile with "warnings as errors", but I didn't want to go so far.

Reply:
Yeah, that's probably just vexing for people who want to use the application on future erlang versions that we haven't adjusted for just yet.

Reply:
Hm... maybe I didn't express myself correctly. I was stating that, e.g., `rebar3` profile 'test' could be "enhanced" with option `{erl_opts, [warnings_as_errors]}` (which I did locally for testing) and that wouldn't break what exists "too much".

In any case, if you're interested, this is what happens when you do that change.

1. compilation will initially stop (first clean run) with
```erlang
test/erlcloud_sdb_tests.erl:4: export_all flag enabled - all functions will be exported
```

2. after that issue is solved compilation will proceed with
```erlang
test/erlcloud_aws_tests.erl:71: this clause cannot match because a previous clause at line 71 always matches
test/erlcloud_aws_tests.erl:72: this clause cannot match because a previous clause at line 72 always matches
test/erlcloud_aws_tests.erl:73: this clause cannot match because a previous clause at line 73 always matches
```

These are the only two issues I'm aware of, for that particular change.

Reply:
I'll deal with this in a separate Pull Request.

Reply:
Makes sense, I didn't realize that you meant only to use warnings_as_errors for unit testing. I am +1 for that change, and I like #650.